core/domain/suggestion_jobs_one_off.py

```diff
@@ -54,6 +55,59 @@ def reduce(key, values):
                 len(values), values))
 
 
+class SuggestionSvgFilenameValidationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks the html content of a suggestion and validates the
+    svg_filename fields in each math rich-text components."""
+
+    _ERROR_KEY = 'found-suggestion-with-invalid-math'
+    _SUCCESS_KEY = 'suggestion-with-latex-strings-having-no-svg'
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        suggestion = suggestion_services.get_suggestion_from_model(item)
+        html_string_list = suggestion.get_all_html_content_strings()
+        html_string = ''.join(html_string_list)
+        if suggestion.target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+            error_list = (
+                html_validation_service.
+                validate_math_tags_in_html_with_attribute_math_content(
+                    html_string))
+            if len(error_list) > 0:
+                yield (
+                    SuggestionSvgFilenameValidationOneOffJob._ERROR_KEY,
+                    item.id)
+                return
+            invalid_tags = (
+                html_validation_service.
+                validate_svg_filenames_in_math_rich_text(
+                    feconf.ENTITY_TYPE_EXPLORATION, item.target_id,
+                    html_string))
+            if len(invalid_tags) > 0:
+                yield (
+                    SuggestionSvgFilenameValidationOneOffJob._SUCCESS_KEY,
+                    (item.id, invalid_tags))
```

Comment:
We don't need to yield the else case here, because we are only concerned about the suggestions having math-tags with no SVGs.
Also in the else case, it is not necessary that the suggestion will have math-tags.